esp-bootloader-esp-idf/src/ota.rs

```diff
@@ -148,62 +140,83 @@ impl OtaSelectEntry {
-/// supported by the ESP-IDF bootloader.
+/// If you are looking for a more high-level way to do this, see [crate::ota_updater::OtaUpdater]
 #[derive(Debug)]
 #[cfg_attr(feature = "defmt", derive(defmt::Format))]
 pub struct Ota<'a, F>
 where
     F: embedded_storage::Storage,
 {
     flash: &'a mut FlashRegion<'a, F>,
+    ota_partition_count: usize,
 }
 
 impl<'a, F> Ota<'a, F>
 where
     F: embedded_storage::Storage,
 {
-    /// Create a [Ota] instance from the given [FlashRegion]
+    /// Create a [Ota] instance from the given [FlashRegion] and the count of OTA app partitions
+    /// (not including "firmware" and "test" partitions)
     ///
     /// # Errors
-    /// A [crate::partitions::Error::InvalidPartition] if the given flash region
-    /// doesn't represent a Data/Ota partition or the size is unexpected
-    pub fn new(flash: &'a mut FlashRegion<'a, F>) -> Result<Ota<'a, F>, crate::partitions::Error> {
+    /// A [Error::InvalidPartition] if the given flash region
+    /// doesn't represent a Data/Ota partition or the size is unexpected.
+    ///
+    /// [Error::InvalidArgument] if the `ota_partition_count` exceeds the maximum or if it's 0.
+    pub fn new(
+        flash: &'a mut FlashRegion<'a, F>,
+        ota_partition_count: usize,
+    ) -> Result<Ota<'a, F>, Error> {
+        if ota_partition_count == 0 || ota_partition_count >= 16 {
+            return Err(Error::InvalidArgument);
+        }
+
         if flash.capacity() != 0x2000
-            || flash.raw.partition_type()
-                != crate::partitions::PartitionType::Data(
-                    crate::partitions::DataPartitionSubType::Ota,
-                )
+            || flash.raw.partition_type() != PartitionType::Data(DataPartitionSubType::Ota)
         {
-            return Err(crate::partitions::Error::InvalidPartition {
+            return Err(Error::InvalidPartition {
                 expected_size: 0x2000,
-                expected_type: crate::partitions::PartitionType::Data(
-                    crate::partitions::DataPartitionSubType::Ota,
-                ),
+                expected_type: PartitionType::Data(DataPartitionSubType::Ota),
             });
         }
 
-        Ok(Ota { flash })
+        Ok(Ota {
+            flash,
+            ota_partition_count,
+        })
     }
 
-    /// Returns the currently active OTA-slot.
-    pub fn current_slot(&mut self) -> Result<Slot, crate::partitions::Error> {
+    /// Returns the currently selected app partition.
+    ///
+    /// This might not be the booted partition if the bootloader failed to boot
+    /// the partition and felt back to the last known working app partition.
+    ///
+    /// See [crate::partitions::PartitionTable::booted_partition] to get the booted
+    /// partition.
+    pub fn current_app_partition(&mut self) -> Result<AppPartitionSubType, Error> {
         let (seq0, seq1) = self.get_slot_seq()?;
 
         let slot = if seq0 == 0xffffffff && seq1 == 0xffffffff {
```

Comment:
Matter of taste: `0xffffffff` is used quite a lot, maybe we can stare it to a variable and use it instead? I guess the same for `0x10`.